WorkloadManagementSystem/Client/DownloadInputData.py

```diff
@@ -92,11 +91,11 @@ def execute(self, dataToResolve=None):
         failedReplicas.add(lfn)
         continue
 
-      # Get and remove size and GUIS
+      # Get and remove size and GUID
       size = reps.pop('Size')
       guid = reps.pop('GUID')
       # Remove all other items that are not SEs
-      for item in reps.keys():
+      for item in list(reps.keys()):
```

Comment:
isn't
```suggestion
      for item in reps:
```
enough?

Reply:
No, there is a pop two lines below

Reply:
Ahh, I only wrote it in the commit message: 80c3855c352d1bbbfd89ac6932c0d0bb694039ce

Reply:
Would this be python 3 safe? 
In any case we need a comment in the code for it.

Reply:
yes it is safe. Will add a comment (`.keys() isn't needed`)
```
2020-07-15 13:56 sailer@lxplus737:~$ cat testPop.py
#!/bin/env python
dct = {'foo': 'bar', 'baz': 'goo'}
try:
  for k in dct:
    dct.pop(k)
except Exception as e:
  print(e)
print(dct)

dct = {'foo': 'bar', 'baz': 'goo'}
try:
  for k in list(dct.keys()):
    dct.pop(k)
except Exception as e:
  print(e)
print(dct)
2020-07-15 13:56 sailer@lxplus737:~$ python3 testPop.py
dictionary changed size during iteration
{'baz': 'goo'}
{}
```

Reply:
Added the comment in the code